application-admintools-default/src/main/java/com/xwiki/admintools/internal/files/resources/XWikiConfigFileDataResource.java

```diff
@@ -0,0 +1,132 @@
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+
+import com.xwiki.admintools.configuration.AdminToolsConfiguration;
+import com.xwiki.admintools.download.DataResource;
+import com.xwiki.admintools.internal.data.identifiers.CurrentServer;
+
+/**
+ * {@link DataResource} implementation for accessing the xwiki.cfg file.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component
+@Named(XWikiConfigFileDataResource.HINT)
+@Singleton
+public class XWikiConfigFileDataResource implements DataResource
+{
+    /**
+     * Component identifier.
+     */
+    public static final String HINT = "xwikiConfig";
+
+    private static final String XWIKI_CFG = "xwiki.cfg";
+
+    private static final String ERROR_SOURCE = " Root cause is: [{}]";
+
+    @Inject
+    @Named("default")
+    private AdminToolsConfiguration adminToolsConfig;
+
+    @Inject
+    private CurrentServer currentServer;
+
+    @Inject
+    private Logger logger;
+
+    @Override
+    public void addZipEntry(ZipOutputStream zipOutputStream, Map<String, String> filters) throws Exception
+    {
+        addZipEntry(zipOutputStream);
+    }
+
+    @Override
+    public byte[] getByteData(String input) throws Exception
+    {
+        try {
+            List<String> excludedLinesHints = adminToolsConfig.getExcludedLines();
+            String filePath = currentServer.getCurrentServer().getXwikiCfgFolderPath() + XWIKI_CFG;
+            File inputFile = new File(filePath);
+            try (BufferedReader reader = new BufferedReader(new FileReader(inputFile))) {
+                StringBuilder stringBuilder = new StringBuilder();
+                String currentLine;
+
+                // Read line by line and do not add it if it contains sensitive info.
+                while ((currentLine = reader.readLine()) != null) {
+                    String trimmedLine = currentLine.trim();
+                    if (excludedLinesHints.stream().anyMatch(trimmedLine::contains)) {
+                        continue;
+                    }
+                    stringBuilder.append(currentLine).append(System.getProperty("line.separator"));
+                }
+                reader.close();
+                return stringBuilder.toString().getBytes();
+            }
+        } catch (IOException exception) {
+            throw new IOException(String.format("Error while handling %s file.", XWIKI_CFG), exception);
+        } catch (NullPointerException e) {
+            logger.warn("Server not found. Root cause is: [{}]", ExceptionUtils.getRootCauseMessage(e));
+            throw new NullPointerException("Server not found.");
+        } catch (RuntimeException e) {
+            String errMessage = "Error while retrieving data from Admin Tools configuration.";
+            logger.warn(errMessage + ERROR_SOURCE, ExceptionUtils.getRootCauseMessage(e));
+            throw new RuntimeException(errMessage, e);
```

Comment:
Same, here you are sending the error message along with the exception, so you don't need to log it in here. The method that really handles this exception will need to get the root cause and log it
Also, RuntimeException is still too generic, it does not give enough information on what went wrong. Both the message and the exception need to be more specific